packages/store/addon/-private/system/core-store.ts

```diff
@@ -2242,16 +2242,24 @@ abstract class CoreStore extends Service {
 
     let snapshotArray = array._createSnapshot(options);
 
-    if (adapter.shouldReloadAll(this, snapshotArray)) {
-      set(array, 'isUpdating', true);
-      return promiseArray(_findAll(adapter, this, modelName, options));
+    if (options.reload !== false) {
+      if (
+        (adapter.shouldReloadAll && adapter.shouldReloadAll(this, snapshotArray)) ||
+        (!adapter.shouldReloadAll && snapshotArray.length > 0)
```

Comment:
The length check here is reversed: We should default to reloading only if the length is `0`. This is what ensures we fire at least one request so long as you've never loaded a record previously by another means. This is a bit of a troll (and always has been) but is the current behavior. We could improve this in the future by tracking whether we've ever made *at least* one findAll request and using that to determine whether we need to make this initial request or not.

```suggestion
        (!adapter.shouldReloadAll && snapshotRecordArray.length === 0)
```

Reply:
This is weird, but nice catch. I'll have to fix tests also.

Reply:
fixed.